### DISCUSSION TITLE
Immediate blocks

### DISCUSSION THREAD

**Discussion by kyouko-taiga** [hylo-lang/hylo#743] [Language design]
It'd be interesting to have a feature that lets us express immediately-invoked lambdas.

```val
public fun main() {
  let x = ...
  let y = do {
    execute_some()
    statements(perhaps_using: x)
    return some_value()
  }
}
```

**Comment by tothambrus11** [comment]
I like the idea. In addition, we could experiment with explicit captures. Historically, there has been a tradeoff between "you should make that its own named function" and the code becoming a sapghetti, where one needs to have an overview of the call graph to understand what's going on.
